cjsmin/src/chess.ts

```diff
@@ -691,6 +692,23 @@ export class Chess {
     this._history = [];
     this._header = keepHeaders ? this._header : {};
     this._updateSetup(this.fen());
+    this._positionCounts = new Proxy({} as Record<string, number>, {
+      get: (target, position: string) =>
+        position === 'length'
+          ? Object.keys(target).length // length for unit testing
+          : target?.[this._trimFen(position)] || 0,
+      set: (target, position: string, count: number) => {
+        const trimmedFen = this._trimFen(position)
+        if (count === 0) delete target[trimmedFen]
+        else target[trimmedFen] = count
+        return true
+      },
+    })
+  }
+
+  private _trimFen(fen: string): string {
+    // remove last two fields in FEN string as they're not needed when checking for repetition
+    return fen.split(' ').slice(0, 4).join(' ')
```

Comment:
can we take all this stuff out? We aren't using it right?

Reply:
Yes. Removed all positionCounts and related changes from latest: https://github.com/bennyrubanov/chessanalysis/pull/52